### PR TITLE
I18n: Add instantiable I18nLocale class 

### PR DESCRIPTION
**Pull request by sirreal**
## Description

Addresses part of https://github.com/WordPress/gutenberg/issues/19210

Adds `I18nLocale` class to the `@wordpress/i18n` package. This change is fully backwards compatible as demonstrated by the passing unchanged tests: https://travis-ci.com/WordPress/gutenberg/jobs/289254678

```js
import { I18nLocale } from '@wordpress/i18n';
const i18n = new I18nLocal();
i18n.__( 'Hello, world!' );
```

## How has this been tested?

Includes unit tests. Also manually tested via Node.js on the command line:

```js
const { I18nLocale } = require('@wordpress/i18n')
const id = new I18nLocale();
id.__('Hello') // 'Hello'
const esLocale = new I18nLocale( { 'Hello, world!': [ 'Hola, mundo!' ] } );
esLocale.__('Hello, world!') // 'Hola, mundo!'
```

## Types of changes

New feature - Add `I18nLocale` class to `@wordpress/i18n` package.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/i18n/README.md [hunk 1]
```diff
@@ -61,79 +55,19 @@ _Returns_
...
-
--   _single_ `string`: The text to be used if the number is singular.
--   _plural_ `string`: The text to be used if the number is plural.
--   _number_ `number`: The number to compare against to use either the singular or plural form.
--   _domain_ `[string]`: Domain to retrieve the translated text.
-
-_Returns_
-
--   `string`: The translated singular or plural form.
+Undocumented declaration.
```
Comment (sirreal) [inline packages/i18n/README.md]: It's unfortunate that the doc parser can't find these definitions. If we decide to move ahead with this we should be able to find a way around this.

**Comment by sirreal** [comment]
Alternate to #20318.

**Comment by jsnajdr** [comment]
@jameslnewell was first: #20318 🙂 

Can you compare the two patches and select the better side of each place where they differ?

**Comment by sirreal** [comment]
> @jameslnewell was first: #20318 🙂
> 
> Can you compare the two patches and select the better side of each place where they differ?

Closing in favor of #20318

I suspect we'll find the code is the same, mostly just moved around, but this structure will tree shake better. @jameslnewell I'll let you bring in what's worthwhile from here.

**Pull request closed without merging**